src/main.rs

```diff
@@ -55,15 +59,27 @@ fn main() -> Result<(), Box<dyn Error>> {
         lon_max,
     };
     let chunks = bounding_box.chunkify(cpus);
+    let completed = Arc::new(AtomicUsize::new(0));
```

Comment:
this actually ended up being a bit more advanced than i thought it would: an _atomic_ integer is a special kind of integer that can be mutated from multiple threads concurrently. we have to use an atomic to implement a shared counter. this is kind of advanced stuff you aren't really expected to have learnt by now at all, but i thought it was worth having a progress counter, so i wanted to add it anyway.